crates/rpc/rpc-api/src/reth.rs

```diff
@@ -0,0 +1,15 @@
+use jsonrpsee::{core::RpcResult, proc_macros::rpc};
+use reth_primitives::{Address, BlockId, U256};
+use std::collections::HashMap;
+
+/// Reth API namespace for reth-specific methods
+#[cfg_attr(not(feature = "client"), rpc(server, namespace = "reth"))]
+#[cfg_attr(feature = "client", rpc(server, client, namespace = "reth"))]
+pub trait RethApi {
+    /// Returns all ETH balance changes in a block
+    #[method(name = "getBalancesChangesInBlock")]
+    fn reth_get_balance_changes_in_block(
+        &self,
+        block_id: BlockId,
+    ) -> RpcResult<Option<HashMap<Address, Option<U256>>>>;
```

Comment:
Should we return old and new balance?

Reply:
Personally yes I prefer prefer returning both; it's not the same thing that erigon returns though, but they have access to the both balances in their code so could update as well.  Since this is in the `reth_` namespace IMO we have leeway to differ w/ their implementation, and of course we can PR a execution-api schema that includes both old and new balances since I don't see the RPC defined over there yet.

I'll get everything else working/tested before coming back to this.

Reply:
I'm going to leave this compatible w/ erigon for now, but definitely think we should argue for a format that contains the old and new balances over in execution-apis.